zerver/webhooks/linear/view.py

```diff
@@ -0,0 +1,167 @@
+    return COMMENT_REMOVE_TEMPLATE.format(user=payload["data"]["user"]["name"].tame(check_string))
+
+
+def get_issue_or_sub_issue_message(payload: WildValue, event: str) -> str:
+    action = payload["action"].tame(check_string)
+    if action == "remove":
+        return get_issue_remove_body(payload, event)
+
+    return get_issue_created_or_updated_message(
+        event, payload, action="created" if action == "create" else "updated"
+    )
+
+
+def get_comment_message(payload: WildValue, event: str) -> str:
+    action = payload["action"].tame(check_string)
+    if action == "remove":
+        return get_comment_remove_body(payload, event)
+
+    return get_comment_create_or_update_body(
+        payload, event, "commented" if action == "create" else "updated comment"
+    )
+
+
+EVENT_FUNCTION_MAPPER: Dict[str, Callable[[WildValue, str], str]] = {
+    "issue": get_issue_or_sub_issue_message,
+    "sub_issue": get_issue_or_sub_issue_message,
+    "comment": get_comment_message,
+}
+
+IGNORED_EVENTS = ["IssueLabel", "Project", "ProjectUpdate", "Cycle", "Reaction"]
+
+ALL_EVENT_TYPES = list(EVENT_FUNCTION_MAPPER.keys())
+
+
+@webhook_view("Linear", notify_bot_owner_on_invalid_json=True, all_event_types=ALL_EVENT_TYPES)
+@has_request_variables
+def api_linear_webhook(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    payload: WildValue = REQ(argument_type="body", converter=to_wild_value),
+    user_specified_topic: Optional[str] = REQ("topic", default=None),
+) -> HttpResponse:
+    event_type = get_event_type(payload)
+    if event_type is None:
+        return json_success(request)
+
+    subject = get_subject(user_specified_topic, event_type, payload)
+
+    body_function = EVENT_FUNCTION_MAPPER[event_type]
+    body = body_function(payload, event_type)
+
+    check_send_webhook_message(request, user_profile, subject, body)
+
+    return json_success(request)
+
+
+def get_subject(user_specified_topic: Optional[str], event: str, payload: WildValue) -> str:
+    if user_specified_topic is not None:
+        return user_specified_topic
+    elif event == "comment":
+        issue_id = payload["data"]["issueId"].tame(check_string)
+        return issue_id
+    elif event == "sub_issue":
+        parent = payload["data"]["parentId"].tame(check_string)
+        return parent
+    elif event == "issue":
+        issue_id = payload["data"]["id"].tame(check_string)
+        return issue_id
+
+    raise UnsupportedWebhookEventTypeError("unknown event type")
+
+
+def get_event_type(payload: WildValue) -> Optional[str]:
+    type = payload["type"].tame(check_string)
+
+    if type == "Issue":
+        has_parent_id = "parentId" in payload["data"]
+        return "issue" if not has_parent_id else "sub_issue"
+    elif type == "Comment":
+        return "comment"
+    elif type in IGNORED_EVENTS:
+        return None
+
+    complete_event = "{}:{}".format(
+        type, payload.get("action", "???").tame(check_string)
+    )  # nocoverage
```

Comment:
Is there any particular reason that we can't achieve coverage here? Might be good to leave a comment.